receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -79,5 +86,13 @@ func (s *scraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	}
 	s.mb.RecordSystemCPULogicalCountDataPoint(now, int64(numCPU))
 
+	cpuInfos, err := s.getCPUInfo()
+	if err != nil {
+		return pmetric.NewMetrics(), scrapererror.NewPartialScrapeError(err, metricsLen)
+	}
+	for _, cInfo := range cpuInfos {
+		s.mb.RecordSystemCPUFrequencyDataPoint(now, cInfo.frequency*hzInAMHz, fmt.Sprintf("cpu%d", cInfo.processor))
+	}
```

Comment:
I'd suggest we additionally check for `s.config.MetricsBuilderConfig.Metrics.SystemCPUFrequency.Enabled` here before calling `getCPUInfo`. Just to avoid throwing the errors if the metric is not enabled